### ISSUE TITLE
Custom lists

### ISSUE THREAD

**Issue by epicmonkey** [pepyatka/pepyatka-server#214]
TBD

**Comment by smallq-git** [comment]
NOTE: there is no spec describing friend list feed behavior, so p. 15 is added based on user's feedback.
Specs used: 
Groups: https://docs.google.com/document/d/1q0-fHGqhmtRihut0X1o1iQYIThFN46pouBoo3-YJgfg/edit#heading=h.gjdgxs
Friends 1: https://docs.google.com/document/d/1u7BxFdzy8f7OfcVm_t0U-CLc35-U4t2ILJQVW4gLNQo/edit#heading=h.477tq198zjp
Friends 2: https://docs.google.com/document/d/1aJGR3v2O2a76U1lcDH9sJBZijeJRrdfjbh2f-pNoMLE/edit

1. On sidebar user is able to see the list of their friend lists and link to manage lists:
![image](https://cloud.githubusercontent.com/assets/7983303/7787528/00c0eba0-021a-11e5-8094-35665ac7de4b.png)
* Friends management page is also available via Friends sidebar item.

2. When subscribing to another user, user gets an option to select existing friend lists this new friend will belong to, as well as create a new friend list for this user:
![image](https://cloud.githubusercontent.com/assets/7983303/7787525/e388bf72-0219-11e5-94b1-562cf864b42f.png)

3. When viewing friend's profile, user can see which lists the friend belongs to and go directly to editing separate lists:
![image](https://cloud.githubusercontent.com/assets/7983303/7787519/a3017930-0219-11e5-816e-d64f26124e21.png)

4. When viewing the list of subscriptions, user can view separate lists and create new friend lists:
![image](https://cloud.githubusercontent.com/assets/7983303/7787530/26d91b28-021a-11e5-8a11-b5186325da5c.png)

5. All lists are available for selection in View dropdown; All Subscriptions is selected by default:
![image](https://cloud.githubusercontent.com/assets/7983303/7787533/4cf4a962-021a-11e5-84ba-e5b9b9978f94.png)
* In Frf four lists existed by default: Home Feed, Personal, Professional and Favorites.

6. If selected list is empty, a prompt to add friends to the list is displayed:
![image](https://cloud.githubusercontent.com/assets/7983303/7787537/6302538a-021a-11e5-9902-771ffa7bdd1b.png)

7. Clicking the prompt brings up friends selection window (with the filtering capabilities - separate issue).
![image](https://cloud.githubusercontent.com/assets/7983303/7787578/8627cfd2-021c-11e5-8b76-ce517b203684.png)

8. When friends are added or removed and list is saved, a yellow information panel is briefly displayed, listing number of friends added or removed:
![image](https://cloud.githubusercontent.com/assets/7983303/7787608/7868a712-021d-11e5-9eb2-7972cc058fb6.png)

9. List editing page can be viewed via URL /friends?list=LIST_NAME (Note options to rename or delete!):
![image](https://cloud.githubusercontent.com/assets/7983303/7787661/f0424110-021f-11e5-9e6d-7365a37046d3.png)

10. When list is selected on the friends page, an option to Add/remove friends appears next to the dropdown:
![image](https://cloud.githubusercontent.com/assets/7983303/7787603/4e41ff56-021d-11e5-9dd2-bf6361a199e2.png)

11. When list is selected on the friends page, only those friends that belong to the list are displayed (with pagination if necessary):
![image](https://cloud.githubusercontent.com/assets/7983303/7787609/905f5c30-021d-11e5-88bd-37f89112a5be.png)
![image](https://cloud.githubusercontent.com/assets/7983303/7787633/be261900-021e-11e5-9926-cd283e23ee27.png)

12. Any additions/deletions to the lists are saved and displayed in Subscription History:
![image](https://cloud.githubusercontent.com/assets/7983303/7787616/bf9387a6-021d-11e5-9e0d-ac7507629d47.png)

13. When New friend list option is selected, first the name of the list is prompted:
![image](https://cloud.githubusercontent.com/assets/7983303/7787645/0625103a-021f-11e5-8309-bb478ca752f0.png)
* Several lists can have the same name. 
* Same window is displayed when renaming a list.
* After that friends selection process is identical to list editing.

14. After list is created and saved, it is displayed on the sidebar and can be accessed via URL /list/LIST_NAME

15. On the friends list page only posts of the friends belonging to that list are displayed. Likes and comments do not appear in the friend list feed.

**Comment by epicmonkey** [comment]
@smallq-git could you remind me what exactly is shown on a list page, e.g. Home feed (which is a pre-defined list, but it is a list as well) shows my posts, my friends posts, posts that they commented on and posts that they liked. Is it the same for any other list?

**Comment by smallq-git** [comment]
@epicmonkey It was not specified in the documentation and I've never used the lists myself. Requesting assistance on ff.

**Comment by epicmonkey** [comment]
Thanks to folks from #pro-description channel. custom lists include only friends posts (which makes home feed really-really special). And this is kinda tricky from UX point of view, i.e. if you move user to a custom list you won't see her likes/comments and as FrF doesn't have a good way to manage likes/comments subscriptions you'll be confused.

**Comment by smallq-git** [comment]
So we need a separate way to track likes/comments subscriptions. We are going to allow to subscribe to them separately from posts, aren't we?

**Comment by zverok** [comment]
> Is it the same for any other list?

Nope. The very idea of lists was "when I put user in list, only his post is shown, not liked/commented by this user".

> After list is created and saved, it is displayed on the sidebar and can be accessed via URL /list/LIST_NAME

For lists with cyrillic names FriendFeed created URLs like /list (for first of them), /list1 and so on. Which is pretty lame, yet seems better than /%D0%96%D0%BE%D0%BF%D0%B0

**Comment by epicmonkey** [comment]
Moving to post-migration. 0.5.0
